src/components/pages/PointSystem/TaskDialog.vue

```diff
@@ -0,0 +1,196 @@
+<template>
+  <dialog-base
+    :visible="visible"
+    @update:visible="handleVisibleChange"
+    :title="pointsForCategory.titleProgress"
+    class="task-dialog"
+  >
+    <div>
+      <p>Complete {{ pointsForCategory.titleProgress }}-related tasks in order to level up your skill</p>
+      <div class="task-dialog__card-progress">
+        <div class="current-level">
+          <p>Your level</p>
+          <p>LVL {{ levelCurrent }} <span>/ LVL 6</span></p>
+        </div>
+        <div class="progress-container">
+          <div class="progress-bar" :style="{ width: progressPercentage + '%' }"></div>
+        </div>
+
+        <div class="next-level">
+          <p>Next level reward</p>
+          <p>{{ pointsForCategory.nextLevelRewardPoints }}</p>
+        </div>
+      </div>
+      <div class="task-dialog__card-current">
+        <div class="img-title">
+          <token-logo v-if="isTokenImage" :token="this.getImageSrc(imageName)" size="small" />
+          <img v-else :src="this.getImageSrc(imageName)" :alt="imageName" />
```

Comment:
Please don't use `this` directly in component template, all that is passed to component render function is already from context. Please fix that in all added components